### PR TITLE
build: add React18 Support

### PR DESCRIPTION
**Pull request by andreasonny83**
Update packages dependencies to include React v18

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: packages/paste-website/src/components/site-wrapper/sidebar/SidebarSeparator.tsx [hunk 1]
```diff
@@ -1,4 +1,3 @@
-import * as React from 'react';
```
Comment (TheSisb) [inline packages/paste-website/src/components/site-wrapper/sidebar/SidebarSeparator.tsx:1]: question: why this change? Usually if a file contains JSX it's common practice to import React even if isn't actively used for certain build tools / bundling purposes. Just curious what error you were seeing to remove these and from where

File: packages/paste-core/components/data-grid/src/utils/cell-management.ts [hunk 2]
```diff
@@ -55,7 +55,7 @@ export const isCell = (element: Element): boolean => {
 export const getClosestCellFrom = (element: Element, dataGridId: string): HTMLElement | null => {
   if (element == null) return null;
   if (!dataGridId) return null;
-  return element.closest(`#${dataGridId} th, #${dataGridId} td`);
+  return element.closest(`[id="${dataGridId}"] th, [id="${dataGridId}"] td`);
```
[the remark this answers is not in the document]
Comment (adamb-92) [reply packages/paste-core/components/data-grid/src/utils/cell-management.ts:58]: element.closest struggles with the formatting of the ids that the useID hook creates. This is a workaround that was found for useID - https://stackoverflow.com/questions/73894096/react-useid-creates-invalid-selector

File: packages/paste-core/components/tooltip/__test__/index.spec.tsx [hunk 3]
```diff
@@ -24,7 +24,7 @@ describe('Tooltip', () => {
     );
     it('should render a tooltip button with aria attributes', () => {
       const renderedTooltipButton = screen.getByRole('button');
-      expect(renderedTooltipButton.getAttribute('aria-describedby')).toEqual('paste-tooltip-1');
+      expect(renderedTooltipButton.getAttribute('aria-describedby')).toEqual('paste-tooltip-:r0:');
```
[the remark this answers is not in the document]
Comment (adamb-92) [reply packages/paste-core/components/tooltip/__test__/index.spec.tsx:27]: This _will_ break with the react 17 tests. This was pushed as part of the fix to the useId library and and to initially get it passing with react 18. There will be a subsequent PR where we will need to change the assertion depending on the react version

File: packages/paste-libraries/uid/src/index.tsx [hunk 4]
```diff
@@ -13,7 +13,8 @@ const useUID = maybeReactUseId !== undefined ? maybeReactUseId : useUIDLegacy;
 const useUIDSeed =
   maybeReactUseId !== undefined
     ? () => {
-        return (seed: string) => `${maybeReactUseId()}-${seed}`;
+        const id = maybeReactUseId();
+        return (seed: string) => `${id}-${seed}`;
```
[the remark this answers is not in the document]
Comment (adamb-92) [reply packages/paste-libraries/uid/src/index.tsx:17]: We noticed the functionality for useUIDSeed was simpler than we initially thought. Rather than generating a new id every time the return function is called, it is only meant to generate one id which is consistent with every call of the return method. Only thing that changes is the suffix that gets passed as an argument.

File: packages/paste-website/src/components/site-wrapper/sidebar/SidebarSeparator.tsx (hunk 1 above)
Comment (eyerean) [reply packages/paste-website/src/components/site-wrapper/sidebar/SidebarSeparator.tsx:1]: tsx check ignored here: https://github.com/twilio-labs/paste/pull/2945
